test/kotlin/integration/proxying/PerformHTTPSRequestBadHostname.kt

```diff
@@ -0,0 +1,105 @@
+package test.kotlin.integration.proxying
+
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.ProxyInfo
+import androidx.test.core.app.ApplicationProvider
+
+import io.envoyproxy.envoymobile.LogLevel
+import io.envoyproxy.envoymobile.Custom
+import io.envoyproxy.envoymobile.Engine
+import io.envoyproxy.envoymobile.UpstreamHttpProtocol
+import io.envoyproxy.envoymobile.AndroidEngineBuilder
+import io.envoyproxy.envoymobile.RequestHeadersBuilder
+import io.envoyproxy.envoymobile.RequestMethod
+import io.envoyproxy.envoymobile.ResponseHeaders
+import io.envoyproxy.envoymobile.StreamIntel
+import io.envoyproxy.envoymobile.engine.JniLibrary
+
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.robolectric.RobolectricTestRunner
+
+//                                                ┌──────────────────┐
+//                                                │   Proxy Engine   │
+//                                                │ ┌──────────────┐ │
+// ┌─────────────────────────┐                  ┌─┼─►listener_proxy│ │
+// │https://api.lyft.com/ping│  ┌──────────────┬┘ │ └──────┬───────┘ │ ┌────────────┐
+// │         Request         ├──►Android Engine│  │        │         │ │api.lyft.com│
+// └─────────────────────────┘  └──────────────┘  │ ┌──────▼──────┐  │ └──────▲─────┘
+//                                                │ │cluster_proxy│  │        │
+//                                                │ └─────────────┴──┼────────┘
+//                                                │                  │
+//                                                └──────────────────┘
+@RunWith(RobolectricTestRunner::class)
+class PerformHTTPSRequestUsingProxy {
+  init {
+    JniLibrary.loadTestLibrary()
+  }
+
+  @Test
+  fun `performs an HTTPs request through a proxy`() {
```

Comment:
let's rename `performs an HTTPs request through a proxy using an async DNS resolution that fails`